programs/drift/src/instructions/user.rs

```diff
@@ -1362,6 +1372,137 @@ pub fn handle_place_and_make_perp_order<'c: 'info, 'info>(
     Ok(())
 }
 
+pub fn place_swift_taker_order<'c: 'info, 'info>(
+    taker_key: Pubkey,
+    taker: &mut RefMut<User>,
+    taker_order_params_message: SwiftOrderParamsMessage,
+    ix_sysvar: &AccountInfo<'info>,
+    sig: [u8; 64],
+    perp_market_map: &PerpMarketMap,
+    spot_market_map: &SpotMarketMap,
+    oracle_map: &mut OracleMap,
+    state: &State,
+) -> Result<bool> {
+    // Authenticate the signature
+    let ix_idx = load_current_index_checked(ix_sysvar)?;
+    validate!(
+        ix_idx > 0,
+        ErrorCode::InvalidVerificationIxIndex,
+        "instruction index must be greater than 0"
+    )?;
+    let ix: Instruction = load_instruction_at_checked(ix_idx as usize - 1, ix_sysvar)?;
+    verify_ed25519_ix(
+        &ix,
+        &taker.authority.to_bytes(),
+        &taker_order_params_message.clone().try_to_vec()?,
+        &sig,
+    )?;
+
+    // Verify inputs and validate market index
+    taker_order_params_message.verify_all_same_market_indexes()?;
+    let market_index = taker_order_params_message.market_index;
+    let taker_order_params = taker_order_params_message.swift_order_params;
+
+    let clock = &Clock::get()?;
+    controller::repeg::update_amm(market_index, perp_market_map, oracle_map, state, clock)?;
```

Comment:
i think we should keep perp market read only for this ix and we can get rid of this fn call